File: questModule/src/main/java/ru/nekit/android/qls/quest/base/Quest.java
package ru.nekit.android.qls.quest.base;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import ru.nekit.android.qls.quest.QuestType;
import ru.nekit.android.qls.quest.QuestionType;

public abstract class Quest<AnswerType> {

    @Nullable
    protected transient Object[] mAvailableAnswerVariants;
    private QuestionType mQuestionType;
    private QuestType mQuestType;

    @NonNull
    public QuestionType getQuestionType() {
        return mQuestionType;
    }

    public void setQuestionType(@NonNull QuestionType questionType) {
        mQuestionType = questionType;
    }

    public QuestType getQuestType() {
        return mQuestType;
    }

    public void setQuestType(@NonNull QuestType questType) {
        mQuestType = questType;
    }

    @Nullable
    public Object[] getAvailableAnswerVariants() {
        return mAvailableAnswerVariants;
    }

    public void setAvailableAnswerVariants(@NonNull Object[] values) {
        mAvailableAnswerVariants = values;
    }

    public abstract AnswerType getAnswer();

    public abstract Class getAnswerClass();

    public abstract int getAnswerInputType();

}

File: questDomain/src/main/java/ru/nekit/android/qls/domain/model/quest/Quest.kt
package ru.nekit.android.qls.domain.model.quest

import ru.nekit.android.qls.shared.model.QuestType
import ru.nekit.android.qls.shared.model.QuestionType

abstract class Quest {

    lateinit var questType: QuestType
    lateinit var questionType: QuestionType
    @Transient
    var availableAnswerVariants: List<Any>? = null
    abstract val answer: Any
    abstract val answerClass: Class<*>

}